src/Kestrel.Core/Adapter/ListenOptionsConnectionLoggingExtensions.cs

```diff
@@ -30,6 +30,11 @@ public static ListenOptions UseConnectionLogging(this ListenOptions listenOption
         public static ListenOptions UseConnectionLogging(this ListenOptions listenOptions, string loggerName)
         {
             var loggerFactory = listenOptions.KestrelServerOptions.ApplicationServices.GetRequiredService<ILoggerFactory>();
+            return listenOptions.UseConnectionLogging(loggerFactory, loggerName);
+        }
+
+        internal static ListenOptions UseConnectionLogging(this ListenOptions listenOptions, ILoggerFactory loggerFactory, string loggerName)
```

Comment:
I didn't want to create a new public API here, but having the connection logging in the test I was tracing seemed somewhat useful. I'm open to this being public, as it doesn't seem to hurt, but I don't really know the public use-case for it :). I needed to add it because in the tests we create `ListenOptions` which do not have an associated `KestrelServerOptions`.

Reply:
Perhaps pubternal instead? (I'm sure @davidfowl has an opinion on this.)

Reply:
It can't really be pubternal because it's an extension method. We could put it in a separate namespace, but we generally put extension methods like this in the same namespace as the type being extended.

Reply:
Instead of adding an extension method, I'd just call `listenOptions.ConnectionAdapters.Add(new LoggingConnectionAdapter(logger));` directly from the test.